core/src/main/java/org/apache/druid/java/util/common/StringUtils.java

```diff
@@ -360,4 +360,93 @@ public static String encodeBase64String(byte[] input)
   {
     return BASE64_DECODER.decode(input);
   }
+
+  /**
+   * Returns the string left-padded with the string pad to a length of len characters.
+   * If str is longer than len, the return value is shortened to len characters. 
+   *
+   * @param base The base string to be padded
+   * @param len The length of padded string
+   * @param pad The pad string
+   * @return the string left-padded with pad to a lenght of len
+   */
+  public static String lpad(String base, Integer len, String pad)
```

Comment:
I believe guava and commons-lang also have string padding functions, how do these compare to those implementations? (Not a suggestion to use either of those, just curious about the choice)

Reply:
@clintropolis I looked at guava and commons-lang3 StringUtils, it has different behavior from expected for length less than base string, such as `lpad("bat", 1, "yz")` expected `b` but commons-lang3 gave `bat`. And guava only pad single char. This method is migrated from Flink's [scala method](https://github.com/apache/flink/blob/master/flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala#L157)

Reply:
Thanks for the explanation, seems reasonable :+1: